packages/components/src/internal/app/utils.ts

```diff
@@ -222,57 +224,223 @@ export function getMenuSectionConfigs(user: User, currentApp: string): List<Map<
+}
+
+function getWorkflowSectionConfig(appBase: string): MenuSectionConfig {
+    return new MenuSectionConfig({
+        headerURL: appBase + WORKFLOW_HOME_HREF.toHref(),
+        iconURL: imageURL('_images', 'workflow.svg'),
+        seeAllURL: appBase + AppURL.create(WORKFLOW_KEY).toHref(),
+    });
+}
+
+function getNotebooksSectionConfig(appBase: string): MenuSectionConfig {
+    return new MenuSectionConfig({
+        iconURL: imageURL('biologics/images', 'notebook_blue.svg'),
+        seeAllURL: appBase + AppURL.create(NOTEBOOKS_KEY).toHref(),
+    });
+}
+
+function getMediaSectionConfig(appBase: string): MenuSectionConfig {
+    return new MenuSectionConfig({
+        headerURL: appBase + AppURL.create(MEDIA_KEY).toHref(),
+        iconURL: imageURL('_images', 'mixtures.svg'),
+        seeAllURL: appBase + AppURL.create(MEDIA_KEY).toHref(),
+    });
+}
+
+function getRegistrySectionConfig(appBase: string): MenuSectionConfig {
+    return new MenuSectionConfig({
+        iconURL: imageURL('_images', 'molecule.svg'),
+        seeAllURL: appBase + AppURL.create(REGISTRY_KEY).toHref(),
+    })
+}
 
-    if (isSampleManagerEnabled()) {
+const USER_SECTION_CONFIG = new MenuSectionConfig({
+    iconCls: 'fas fa-user-circle ',
+});
+
+const REQUESTS_SECTION_CONFIG = new MenuSectionConfig({
+    headerURL: buildURL('query', 'executeQuery', {
+        schemaName: 'issues',
+        'query.queryName': 'IssueListDef',
+    }),
+    iconURL: imageURL('_images', 'default.svg'),
+});
+
+
+// exported for testing
+export function getMenuSectionConfigs(user: User, currentProductId: string, moduleContext?: any): List<Map<string, MenuSectionConfig>> {
+    let sectionConfigs = List<Map<string, MenuSectionConfig>>();
+
+    const appBase = getApplicationUrlBase(getPrimaryAppProperties(moduleContext).moduleName, currentProductId, moduleContext);
+    const currentAppProperties = getCurrentAppProperties(); // based on the controller name
+    const isSMPrimary = sampleManagerIsPrimaryApp(moduleContext);
+    const isBioPrimary = biologicsIsPrimaryApp(moduleContext);
+    const isBioOrSM = isSMPrimary || isBioPrimary;
+    if (isSMPrimary || currentAppProperties?.productId === SAMPLE_MANAGER_APP_PROPERTIES.productId) {
+        sectionConfigs = addSourcesSectionConfig(user, appBase, sectionConfigs);
+    }
+    else if (isBioPrimary ) {
+        sectionConfigs = sectionConfigs.push(  Map({ [REGISTRY_KEY]: getRegistrySectionConfig(appBase) }));
+    }
+    if (isBioOrSM) {
+        sectionConfigs = addSamplesSectionConfig(user, appBase, sectionConfigs);
+        sectionConfigs = addAssaysSectionConfig(user, appBase, sectionConfigs);
+    }
+
+    const storageConfig = getStorageSectionConfig(user, currentProductId, moduleContext, isBioPrimary && isRequestsEnabled(moduleContext) ? 7 : 12);
+    const workflowConfig = getWorkflowSectionConfig(appBase);
+
+    if (isSMPrimary || currentAppProperties?.productId === SAMPLE_MANAGER_APP_PROPERTIES.productId) {
```

Comment:
this if clause is used a couple of times, may be worth putting into a variable so that the name can become documentation for "we are in the SM app"

Reply:
Good thought.  Added.